lib/go-atscfg/meta.go

```diff
@@ -20,102 +20,148 @@ package atscfg
+	} else if server.ProfileID == nil {
+		return nil, warnings, errors.New("server missing ProfileID")
+	} else if server.TCPPort == nil {
+		return nil, warnings, errors.New("server missing TCPPort")
+	} else if server.HostName == nil {
+		return nil, warnings, errors.New("server missing HostName")
+	} else if server.CDNID == nil {
+		return nil, warnings, errors.New("server missing CDNID")
+	} else if server.CDNName == nil {
+		return nil, warnings, errors.New("server missing CDNName")
+	} else if server.ID == nil {
+		return nil, warnings, errors.New("server missing ID")
+	} else if server.Profile == nil {
+		return nil, warnings, errors.New("server missing Profile")
 	}
-}
 
-func MakeMetaConfig(
-	server *tc.ServerNullable,
-	tmURL string, // global tm.url Parameter
-	tmReverseProxyURL string, // global tm.rev_proxy.url Parameter
-	locationParams map[string]ConfigProfileParams, // map[configFile]params; 'location' and 'URL' Parameters on serverHostName's Profile
-	uriSignedDSes []tc.DeliveryServiceName,
-	scopeParams map[string]string, // map[configFileName]scopeParam
-	dses map[tc.DeliveryServiceName]tc.DeliveryServiceNullableV30,
-	cacheGroupArr []tc.CacheGroupNullable,
-	topologies []tc.Topology,
-) string {
-	configDir := "" // this should only be used for Traffic Ops, which doesn't have a local ATS install config directory (and thus will fail if any location Parameters are missing or relative).
-	return MetaObjToMetaConfig(MakeMetaObj(server, tmURL, tmReverseProxyURL, locationParams, uriSignedDSes, scopeParams, dses, cacheGroupArr, topologies, configDir))
-}
+	tmURL, tmReverseProxyURL := getTOURLAndReverseProxy(globalParams)
+	if tmURL == "" {
+		warnings = append(warnings, "global tm.url parameter missing or empty! Setting empty in meta config!")
+	}
 
-func MetaObjToMetaConfig(atsData tc.ATSConfigMetaData, err error) string {
-	if err != nil {
-		return "error creating meta config: " + err.Error()
+	dses, dsWarns := filterConfigFileDSes(server, deliveryServices, deliveryServiceServers)
+	warnings = append(warnings, dsWarns...)
+
+	locationParams := getLocationParams(serverParams)
+
+	uriSignedDSes, signDSWarns := getURISignedDSes(dses)
+	warnings = append(warnings, signDSWarns...)
+
+	configFiles := []CfgMeta{}
+
+	if locationParams["remap.config"].Path != "" {
+		configLocation := locationParams["remap.config"].Path
+		for _, ds := range uriSignedDSes {
+			cfgName := "uri_signing_" + string(ds) + ".config"
+			// If there's already a parameter for it, don't clobber it. The user may wish to override the location.
+			if _, ok := locationParams[cfgName]; !ok {
+				p := locationParams[cfgName]
+				p.Name = cfgName
+				p.Path = configLocation
+				locationParams[cfgName] = p
+			}
+		}
 	}
-	bts, err := json.Marshal(atsData)
-	if err != nil {
-		// should never happen
-		log.Errorln("marshalling meta config: " + err.Error())
-		bts = []byte("error encoding to json, see log for details")
+
+locationParamsFor:
```

Comment:
To `continue` and `break` out of the outer loop, from the inner loop:
https://github.com/apache/trafficcontrol/blob/c28a6dc95577796bdefa0da935ab520dcf08f63f/lib/go-atscfg/meta.go#L113
https://github.com/apache/trafficcontrol/blob/c28a6dc95577796bdefa0da935ab520dcf08f63f/lib/go-atscfg/meta.go#L115
The `prefixFor` label isn't technically necessary, but I think it makes it clearer, since there are multiple loops we're continuing and breaking from.

Reply:
oh, I see.